### DISCUSSION TITLE
Overhaul RTSP

### DISCUSSION THREAD

**Discussion by Nemobi** [arendst/Tasmota#15690] [Ideas]
I've been looking at the RTSP codebase and I think there's big room for improvements here. 
Much of the code is really only useful in a standalone example, like the code that initalizes the cam which is already handled by the webcam driver. The rtp stream is also currently capped at 10fps which is odd since rtp should performa comparable to mjpeg. Probably better since the stream is over udp so temporary network issues and a few dropped packets won't block the worker thread, and theres less overhead.

So I'm thinking of stripping it down to the bare minimum for providing RTSP support and tiding it upp a bit. I'll probably try and add authentication while I'm at it.

**Comment by s-hadinger** [comment]
You're more than welcome to contribute. This part of the code was mostly untouched for a while.
